src/modules/kvs/kvs.c

```diff
@@ -2262,23 +2259,7 @@ static int namespace_create (struct kvs_ctx *ctx, const char *ns,
         return -1;
     }
 
-    if (!(rootdir = treeobj_create_dir ())) {
-        flux_log_error (ctx->h, "%s: treeobj_create_dir", __FUNCTION__);
-        goto cleanup;
-    }
-
-    if (!(data = treeobj_encode (rootdir))) {
-        flux_log_error (ctx->h, "%s: treeobj_encode", __FUNCTION__);
-        goto cleanup;
-    }
-    len = strlen (data);
-
-    if (blobref_hash (ctx->hash_name, data, len, ref, sizeof (ref)) < 0) {
-        flux_log_error (ctx->h, "%s: blobref_hash", __FUNCTION__);
-        goto cleanup;
-    }
-
-    setroot (ctx, root, ref, 0);
+    setroot (ctx, root, rootref, 0);
```

Comment:
Just to veirfy - do we know for sure that the object pointed to by the blobref (the empty treeobj) is valid in the content cache?  I'm assuming we must know that because this didn't change from before AFAIKT

Reply:
yes, at the moment its guaranteed b/c that's how the primary KVS is initialized (see `store_initial_rootdir()`).  Although it would be good to add a note about it.  I'll add a comment.